Countly.m

```diff
@@ -81,10 +81,22 @@ - (void)startWithConfig:(CountlyConfig *)config
     CountlyCommon.sharedInstance.shouldIgnoreTrustCheck = config.shouldIgnoreTrustCheck;
     CountlyCommon.sharedInstance.loggerDelegate = config.loggerDelegate;
     CountlyCommon.sharedInstance.internalLogLevel = config.internalLogLevel;
-    CountlyCommon.sharedInstance.maxKeyLength = config.maxKeyLength;
-    CountlyCommon.sharedInstance.maxValueLength = config.maxValueLength;
-    CountlyCommon.sharedInstance.maxSegmentationValues = config.maxSegmentationValues;
-
+    
+    CountlyCommon.sharedInstance.maxKeyLength = config.sdkInternalLimits.getMaxKeyLength;
+    CountlyCommon.sharedInstance.maxValueLength = config.sdkInternalLimits.getMaxValueSize;
+    CountlyCommon.sharedInstance.maxSegmentationValues = config.sdkInternalLimits.getMaxSegmentationValues;
+    
+    // For backward compatibility, deprecated values are only set incase new values are not provided using sdkInternalLimits interface
+    if(CountlyCommon.sharedInstance.maxKeyLength == kCountlyMaxKeyLength && config.maxKeyLength != kCountlyMaxKeyLength) {
+        CountlyCommon.sharedInstance.maxKeyLength = config.maxKeyLength;
+    }
+    if(CountlyCommon.sharedInstance.maxValueLength == kCountlyMaxValueSize && config.maxValueLength != kCountlyMaxValueSize) {
+        CountlyCommon.sharedInstance.maxValueLength = config.maxValueLength;
+    }
+    if(CountlyCommon.sharedInstance.maxSegmentationValues == kCountlyMaxSegmentationValues && config.maxSegmentationValues != kCountlyMaxSegmentationValues) {
+        CountlyCommon.sharedInstance.maxSegmentationValues = config.maxSegmentationValues;
+    }
+    
     CountlyConsentManager.sharedInstance.requiresConsent = config.requiresConsent;
```

Comment:
what about other two limits? Is it accessible through config class?

Reply:
Which other two limits? :)

Reply:
maxStackTraceLinesPerThread and maxStackTraceLineLength

Reply:
Yes, we are printing a warning against them in setter that they are just placeholder